modules/solid_mechanics/include/timeintegrators/CentralDifferenceDirect.h

```diff
@@ -0,0 +1,56 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ExplicitTimeIntegrator.h"
+#include "libmesh/sparse_matrix.h"
+
+class MassMatrixHelper;
+/**
+ * Implements a truly explicit (no nonlinear solve) first-order, forward Euler
+ * time integration scheme.
+ */
+class CentralDifferenceDirect : public ExplicitTimeIntegrator
+{
+public:
+  static InputParameters validParams();
+
+  CentralDifferenceDirect(const InputParameters & parameters);
+
+  virtual int order() override { return 1; }
+  virtual void computeTimeDerivatives() override;
+
+  virtual void solve() override;
+  virtual void postResidual(NumericVector<Number> & residual) override;
+
+  virtual bool performExplicitSolve(SparseMatrix<Number> & mass_matrix) override;
+
+  // Applying pre-set nodal BCs
+  void setPresetBCs();
+
+  void computeADTimeDerivatives(ADReal & ad_u_dot,
+                                const dof_id_type & dof,
+                                ADReal & ad_u_dotdot) const override
+  {
+    ad_u_dotdot.value();
+    ad_u_dot.value();
+    std::cout << dof << std::endl;
+    return mooseError("NOT SUPPORTED");
+  }
```

Comment:
What's going on here?

Reply:
This is a copy of what was found in `CentralDifference.C` to overcome the override requirement for `ComputeADTimeDerivatives`

Reply:
But do you need these three lines?
```
    ad_u_dotdot.value();
    ad_u_dot.value();
    std::cout << dof << std::endl;
```

Reply:
Those are too suppress compiler messages. It's a bit jank but I assumed it was acceptable since it was already implemented.

Reply:
>This is a copy of what was found in CentralDifference.h to overcome the override requirement for ComputeADTimeDerivatives

Not quite sure what you mean here. There are no errors in the AD routine in `CentralDifference` in upstream MOOSE

Reply:
Hmm. you're right. I'm not sure what I mean either. I can't remember where I got this from. 
What is the correct way to handle this case?

Reply:
If you don't want to support AD, you can just make this function a one-line `mooseError`. And then you can delete the function parameters so that you don't get unused variable warnings

Reply:
[ab8b9fe](https://github.com/idaholab/moose/pull/28175/commits/ab8b9fe85692cb185dda7f6264efbf1c75a0e597)